dags/data_cleaning_utils/resume_state.py

```
@@ -0,0 +1,94 @@
+
+    def reset(self) -> bool:
+        """Remove stored resume_index from MongoDB. Returns True on success, False on error."""
+        try:
+            self.collection.update_one(
+                {"_id": ObjectId(self.platform_id)},
+                {
+                    "$unset": {"metadata.resume_index": ""},
+                    "$set": {"updatedAt": datetime.now(tz=timezone.utc)},
+                },
+            )
+            return True
```

Ruff (0.12.2)

Consider moving this statement to an `else` block

(TRY300)